src/twisted/internet/test/test_unix.py

```diff
@@ -382,6 +385,56 @@ def test_fileDescriptorOverrun(self):
         test_fileDescriptorOverrun.skip = sendmsgSkip
 
 
+    def test_multiFileDescriptorReceivedPerRecvmsg(self):
+        """
+        Verify that _SendmsgMixin handles receiving multiple file descriptors
+        per recvmsg, calling L{IFileDescriptorReceiver.fileDescriptorReceived}
+        once per received file descriptor.
+        """
+        # Strategy:
+        # - Create a UNIX socketpair.
+        # - Associate one end to a FakeReceiver and FakeProtocol.
+        # - Call sendmsg on the other end with two FDs as ancillary data.
+        # - Call doRead in the FakeReceiver.
+        # - Verify results on FakeProtocol.
+
+        @implementer(IFileDescriptorReceiver)
+        class FakeProtocol(object):
+            def __init__(self):
+                self.fds = []
+            def fileDescriptorReceived(self, fd):
+                self.fds.append(fd)
+                close(fd)
+
+        class FakeReceiver(_SendmsgMixin):
+            bufferSize = 1024
+            def __init__(self, skt, proto):
+                self.socket = skt
+                self.protocol = proto
+            def _dataReceived(self, data):
+                pass
```

Comment:
`adoptStreamConnect` _might_ be just as easy as this, without the abstraction violation.

https://twistedmatrix.com/documents/15.5.0/api/twisted.internet.interfaces.IReactorSocket.html#adoptStreamConnection